python/ray/tests/test_dynres.py

```diff
@@ -286,14 +286,15 @@ def set_res(resource_name, resource_capacity, res_client_id):
     target_node_id = node_ids[1]
     ray.get(set_res.remote(res_name, res_capacity, target_node_id))
 
-    def check_resources():
-        return ray.cluster_resources().get(res_name, None) == res_capacity
-
-    wait_for_condition(check_resources)
+    wait_for_condition(
+        lambda: ray.cluster_resources().get(res_name, None) == res_capacity)
 
     # Delete the resource
     ray.get(delete_res.remote(res_name, target_node_id))
 
+    wait_for_condition(
```

Comment:
Haven't fully verified it yet, but I suspect the issue was that we need to wait for the update to take effect. That is, `delete_res` happens asynchronously. Is that correct?